### DISCUSSION TITLE
Use in Neovim

### DISCUSSION THREAD

**Discussion by tjdevries** [euclidianAce/ltreesitter#3] [Ideas]
Yo @vigoux should we try and pursue something like this for internal use of treesitter? :) I don't know how any of that works, figured I'd just mention it here.

**Comment by euclidianAce** [comment]
iirc, neovim's internal tree-sitter bindings look kinda similar to these already\*, additionally it uses libUV to open grammars' .so and .dll files whereas I have some ifdefs around `LoadLibrary` and `dlopen`, but as it's open source, neovim is free to steal/use any of this if any of those super cool core-dev people find it neat ;)

\* the only real difference from the glances I've taken are that neovim uses some 5.1-isms (setfenv mostly) to keep references alive, whereas I try to keep this 5.1-5.4 compatible by using the registry and some weak tables

**Comment by vigoux** [comment]
Interesting work!
I'll look into using this for the tree-sitter integration!

Externalising things is always a good thing.

**Comment by euclidianAce** [comment]
@vigoux Do you still have any interest in using this? I've been making some refactors lately to make things more manageable and wanted to start work on the public C-side api for it. I assume neovim would prefer to include it from the C side (as `vim.treesitter` or `vim.ltreesitter` or whatever) and not have ltreesitter be duplicating the contents of entire buffers just to do parsing.

But I know the api's have diverged a little bit and `ltreesitter` doesn't cover the whole api yet so it's completely understandable if you don't. (Also I don't want to be a reason that 0.5 is held back :laughing:)
